legacy_packages/sdk/src/evm/common/any-evm-utils/getCreate2FactoryAddress.ts

```diff
@@ -16,23 +17,75 @@ import { getCreate2FactoryDeploymentInfo } from "./getCreate2FactoryDeploymentIn
 export async function getCreate2FactoryAddress(
   provider: providers.Provider,
 ): Promise<string> {
-  const commonFactoryExists = await isContractDeployed(
-    COMMON_FACTORY,
-    provider,
+  const deploymentInfo = await computeCreate2FactoryTransaction(provider);
+  return deploymentInfo.deployment;
+}
+
+/**
+ * @internal
+ */
+export async function computeCreate2FactoryTransaction(
+  provider: providers.Provider,
+): Promise<KeylessDeploymentInfo> {
+  const networkId = (await provider.getNetwork()).chainId;
+
+  // special handling for chains with hardcoded gasPrice and gasLimit
+  if (CUSTOM_GAS_FOR_CHAIN[networkId]) {
+    const enforceEip155 = await isEIP155Enforced(provider);
+    const chainId = enforceEip155 ? networkId : 0;
+    const gasPrice = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasPrice;
+    const gasLimit = CUSTOM_GAS_FOR_CHAIN[networkId]?.gasLimit;
+
+    const deploymentInfo = getCreate2FactoryDeploymentInfo(chainId, {
+      gasPrice,
+      gasLimit,
+    });
+
+    if (await isContractDeployed(deploymentInfo.deployment, provider)) {
+      deploymentInfo.transaction = "";
+    }
+
+    return deploymentInfo;
+  }
+
```

Comment:
should we do a special case to prioritize the check for mainnet keyless deploy address?

Reply:
no because it will cause a waterfall (because once we check and if its' not there we have to continue)